probe-rs/src/config/target.rs

```diff
@@ -68,15 +68,21 @@ impl Target {
         let mut memory_map = chip.memory_map.clone();
         let mut flash_algorithms = Vec::new();
         for algo_name in chip.flash_algorithms.iter() {
-            let mut algo = family
+            let algo = family
                 .get_algorithm(algo_name)
                 .expect(
                     "The required flash algorithm was not found. This is a bug. Please report it.",
                 )
                 .clone();
 
+            let mut algo_cores = if algo.cores.is_empty() {
```

Comment:
I feel like we should fix that in the algo, not here.

Reply:
Sure but IIRC the raw algo is what serde reads out of the yamls.